src/Sulu/Bundle/AdminBundle/Resources/js/components/CircleSelection/CircleSelection.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+import type {Node} from 'react';
+import React from 'react';
+import withContainerSize from '../withContainerSize';
+import CircleSelectionRenderer from './CircleSelectionRenderer';
+import type {SelectionData} from './types';
+import circleSelectionStyles from './circleSelection.scss';
+
+type Props = {
+    children?: Node,
+    disabled: boolean,
+    filled: boolean,
+    label?: string,
+    maxRadius?: number,
+    minRadius?: number,
+    onChange: (value: ?SelectionData) => void,
+    percentageValues: boolean,
+    resizable: boolean,
+    round: boolean,
+    value: SelectionData | typeof undefined,
+};
+
+class CircleSelection extends React.Component<Props & {
+    containerHeight: number,
+    containerWidth: number,
+}> {
+    static defaultProps = {
+        disabled: false,
+        filled: false,
+        percentageValues: false,
+        resizable: true,
+        round: true,
+    };
+
+    render() {
+        const {
+            children,
+            containerHeight,
+            containerWidth,
+            disabled,
+            filled,
+            label,
+            maxRadius,
+            minRadius,
+            onChange,
+            percentageValues,
+            resizable,
+            round,
+            value,
+        } = this.props;
+
+        return (
+            <div className={circleSelectionStyles.selection}>
+                {children}
+                <CircleSelectionRenderer
+                    containerHeight={containerHeight}
+                    containerWidth={containerWidth}
+                    disabled={disabled}
+                    filled={filled}
+                    label={label}
+                    maxRadius={maxRadius}
+                    minRadius={minRadius}
+                    onChange={onChange}
+                    percentageValues={percentageValues}
+                    resizable={resizable}
+                    round={round}
+                    value={value}
+                />
+            </div>
+        );
+    }
+}
+
+export {
+    CircleSelection,
+};
+
+const Component = withContainerSize(CircleSelection, circleSelectionStyles.container);
+
+export default class CircleSelectionComponent extends React.Component<Props> {
+    static defaultProps = CircleSelection.defaultProps;
+
+    static Renderer = CircleSelectionRenderer;
+
+    render() {
+        return <Component {...this.props} />;
```

Comment:
The `CircleSelectionRenderer` is the circle, the `CircleSelection` renders the circle as sibling of the given children, which is the container ... If you want to use a standalone `CircleSelection` (one single circle in a container), you should use the `CircleSelection`. But the `ImageMap` field type is more complex. It renders multiple circles for one single container, therefore I need a standalone class which just renders the circle and has nothing to do with the container. This is the `CircleSelectionRenderer`

Reply:
And the reason for not using your suggestion is the `static Renderer = CircleSelectionRenderer;`, that wouldn't be possible otherwise

Reply:
Is there any reason we cannot decide based on a prop in a single component in which kind of `mode` (maybe something like that could be the name of the prop, open for better suggestions) the component is running. Would allow us to avoid constructions like this (you said yourself that you are not happy with the `Renderer` name`) and keep logic together that IMO belongs together.

Maybe a better word instead of `mode` would be `hasContainer`? But maybe we can discuss that after you have answered if doing something like that based on a prop is feasible in general.

Reply:
I think that should be possible. Should we maybe completely omit the `withContainerSize` wrapper in this container? Then every developer is responsible to pass `containerWidth` and `containerHeight`, no matter how he get's that. Since the `children` prop of a `CircleSelection` is it's container, we could just omit a prop to determine if it has a container and check, if children are set.

Reply:
Maybe I don't understand that correctly, but the `children` is the sibling of the selection itself, not the container, right?

And why do you want to remove the `withContainerSize`? It worked pretty well for me until now, and contains code that is quite cumbersome and tedious to write IMO.

Reply:
1. yes
2. I just want to remove it from this container. The ImageRectangleSelection passes custom `containerWidth` and `containerHeight` and the `ImageMap` passes custom `containerWidth` and `containerHeight`, so we actually do not need the `withContainerSize` wrapper in this container itself. If someone wants to use the `CircleSelection` on his own, he can just wrap it with `withContainerSize` by himself

Reply:
The `ImageRectangleSelection` only works this way, because you've changed it in this PR. Before this PR it was not necessary to pass these sizes, and I would prefer it to keep this way, since this is another source of error (and we've already had enough issues with this kind of stuff). Unless you have a good reason that it should not work anymore.

Reply:
The `RectangleSelection` changed already, so there is already another source of error for the image cropping. It would be a small change for the `ImageRectangleSelection` leading to a much simpler `RectangleSelection` and `CircleSelection`. So the risk is not higher with the changes in the `ImageRectangleSelection` as without them.

Reply:
Additionally, previosly the `ImageRectangleSelection` and the `RectangleSelection` both wrapped themselves using `withContainerSize` which does also not feel very good

Reply:
I think we should discuss that in person, and add an explanation here afterwards.

Reply:
We decided in our call that we merge the `Renderer` components with both `RectangleSelection` and `CircleSelection`. Furthermore we only use the `withContainerSize` and a container HTML element if some `children` is passed. If no `children` are passed the parent component is responsible for setting `containerHeight` and `containerWidth`. This is because if the `CircleSelection` and `RectangleSelection` component would handle the container elements on their own, only the most top circle or rectangle would be movable. This change (mostly) keeps BC, that's also one of the reasons we decided to do it this way.